asllib/doc/ASLTypingReference.tex

```diff
@@ -538,48 +753,81 @@ \subsection{Prose}
 
 \begin{emptyformal}
     \subsection{Formally}
+
+\[
+\begin{array}{c}
+\inferrule{}{E \entails \unconstrainedinteger : \isunconstrainedinteger}\\
+\\
+\inferrule{}{E \entails \TInt(\langle [] \rangle) : \isunderconstrainedinteger}\\
+\\
+\RuleComment{This is not entirely correct, since the constraints may be}\\
+\RuleComment{malformed, or include under-constrained integers.}\\
+\RuleComment{But this is what I go with, for now.}\\
+\inferrule{}{E \entails \TInt(\langle [c_1,\ldots,c_k]\rangle) : \iswellconstrainedinteger}
+\end{array}
+\]     
+    
+\todocomment{
+ROMAN: see note about domains.
+}
 \end{emptyformal}
 
 \subsection{Comments}
     This aims to encompass LRM Section 3.4.1 D\_ZTPP, R\_WJYH, R\_HJPN, R\_CZTX, R\_TPHR
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \chapter{Type Satisfaction and Related Notions}
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\todocomment{
+It is not obvious to me why sometimes we check for type satisfaction and other times we check the structure of a type.
```

Comment:
yes please unify this as part of this PR

Reply:
Well, we seem to be consistent with the LRM so it's best we ask the ASL workgroup before making any changes.